### DISCUSSION TITLE
Localization of Qute templates with lots of static text

### DISCUSSION THREAD

**Discussion by cdhermann** [quarkusio/quarkus#25781] [Q&A] [answered]
I'm developing a Qute template with lots of static, HTML formatted text and very little of dynamic parts. For the static text I use the MessageBundle feature. For a lot of static text, this isn't quite convenient. Are there localization solutions adressing the following problems?

1. Different template for languages. E.g. `template.de-AT.html`. For Austrian German, automatically this template is selected
2. Different languages do number formatting differently. Is it possible to retrieve the Locale object inside a qute template which is actually used by Qute when rendering the document to be able to use `{str:fmt(locale,'%tA',now)} or decide on time and date formatting?
3. When using `@CheckedTemplate` and `Templates.TemplateName().renderAsync()` how do I specify a different locale than the one defined in the property `quarkus.default-locale`?

[a comment by an automated account is not shown]

**Comment by mkouba** [comment, marked as answer]
> For a lot of static text, this isn't quite convenient.

I agree that message bundles are not a good match for this use case.

>     1. Different template for languages. E.g. `template.de-AT.html`. For Austrian German, automatically this template is selected

This is not supported yet.

>     2. Different languages do number formatting differently. Is it possible to retrieve the Locale object inside a qute template which is actually used by Qute when rendering the document to be able to use `{str:fmt(locale,'%tA',now)} or decide on time and date formatting?

Not out of the box but you can use an extension method like this:

```
@TemplateExtension(namespace = "attr")
static String locale(@TemplateAttribute("locale") Object locale) {
   return locale;
}
```
and then use `{str:fmt(attr:locale,'%tA',now)}` in the template.

I've just created https://github.com/quarkusio/quarkus/issues/25785.


> 
>     3. When using `@CheckedTemplate` and `Templates.TemplateName().renderAsync()` how do I specify a different locale than the one defined in the property `quarkus.default-locale`?

You can set the locale attribute of a template instance as described in the [docs](https://quarkus.io/guides/qute-reference#localization), i.e. something like `Templates.TemplateName().setAttribute("locale", Locale.GERMAN).renderAsync()`. When using Qute together with RESTEeasy then the `locale` attribute is derived from the the `Accept-Language` header if not set manually.

**Reply by cdhermann** [reply]
Thanks a lot for your exhaustive answer. Now, I think I also understand https://quarkus.io/guides/qute-reference#localization. This means when using [quarkus-resteasy-qute](https://quarkus.io/guides/qute-reference#resteasy_integration) Qute doesn't do any auto formatting with respect to the locale. It just sets the locale attribute for the developer as a convenience feature in addition to selecting the right MessageBundle for the locale.
